elide-core/src/main/java/com/yahoo/elide/security/permissions/ExpressionBuilder.java

```diff
@@ -0,0 +1,203 @@
+    private static final BiFunction<Expression, Expression, Expression> ANY_JOINER = OrExpression::new;
+
+    /**
+     * Build an expression that checks a specific field.
+     *
+     * @param resource Resource
+     * @param annotationClass Annotation calss
+     * @param field Field
+     * @param changeSpec Change spec
+     * @param <A> Type parameter
+     * @return Commit and operation expressions
+     */
+    public <A extends Annotation> Expressions buildSpecificFieldExpression(final PersistentResource resource,
+                                                                           final Class<A> annotationClass,
+                                                                           final String field,
+                                                                           final ChangeSpec changeSpec) {
+        ExtractedChecks extracted = new ExtractedChecks(resource, annotationClass, field);
+
+        Expression operationExpression =
+                buildFullExpression(extracted.getEntityChecks(),
+                                    extracted.getFieldChecks(),
+                                    (check) -> new DeferredCheckExpression(check, resource, changeSpec, cache),
+                                    SpecificFieldExpression::new);
+        Expression commitExpression =
+                buildFullExpression(extracted.getEntityChecks(),
+                                    extracted.getFieldChecks(),
+                                    (check) -> new ImmediateCheckExpression(check, resource, changeSpec, cache),
+                                    SpecificFieldExpression::new);
+
+        return new Expressions(operationExpression, commitExpression);
+    }
+
+    /**
+     * Build an expression that checks any field on a bean.
+     *
+     * @param resource Resource
+     * @param annotationClass annotation class
+     * @param changeSpec change spec
+     * @param <A> type parameter
+     * @return Commit and operation expressions
+     */
+    public <A extends Annotation> Expressions buildAnyFieldExpression(final PersistentResource resource,
+                                                                      final Class<A> annotationClass,
+                                                                      final ChangeSpec changeSpec) {
+
+        Function<Check, Expression> deferredCheckBuilder =
+                (check) -> new DeferredCheckExpression(check, resource, changeSpec, cache);
+        Function<Check, Expression> immediateCheckBuilder =
+                (check) -> new ImmediateCheckExpression(check, resource, changeSpec, cache);
+
+        List<String> fields = resource.getDictionary().getAllFields(resource.getObject());
+
+        ExtractedChecks entity = new ExtractedChecks(resource, annotationClass);
+        Expression opExp =
+                buildFullExpression(entity.getEntityChecks(),
+                                    entity.getFieldChecks(),
```

Comment:
What are the field checks that are being built here?  It seems like the building for those actually happens in the for loop below.